Firestore/Swift/Tests/Integration/VectorIntegrationTests.swift

```diff
@@ -0,0 +1,238 @@
+    XCTAssertEqual(
+      snapshot.get("vector2") as? VectorValue,
+      FieldValue.vector([0, 0, 0] as [Double])
+    )
+    XCTAssertEqual(
+      snapshot.get("vector3") as? VectorValue,
+      FieldValue.vector([-1, -200, -999] as [Double])
+    )
+  }
+
+  func testSdkOrdersVectorFieldSameWayAsBackend() async throws {
+    let collection = collectionRef()
+
+    var docsInOrder: [[String: Any]] = [
+      ["embedding": [1, 2, 3, 4, 5, 6]],
+      ["embedding": [100]],
+      ["embedding": FieldValue.vector([Double.infinity * -1])],
+      ["embedding": FieldValue.vector([-100.0])],
+      ["embedding": FieldValue.vector([100.0])],
+      ["embedding": FieldValue.vector([Double.infinity])],
+      ["embedding": FieldValue.vector([1, 2.0])],
+      ["embedding": FieldValue.vector([2, 2.0])],
+      ["embedding": FieldValue.vector([1, 2, 3.0])],
+      ["embedding": FieldValue.vector([1, 2, 3, 4.0])],
+      ["embedding": FieldValue.vector([1, 2, 3, 4, 5.0])],
+      ["embedding": FieldValue.vector([1, 2, 100, 4, 4.0])],
+      ["embedding": FieldValue.vector([100, 2, 3, 4, 5.0])],
+      ["embedding": ["HELLO": "WORLD"]],
+      ["embedding": ["hello": "world"]],
+    ]
+
+    var docs: [[String: Any]] = []
+    for data in docsInOrder {
+      let docRef = try await collection.addDocument(data: data)
+      docs.append(["id": docRef.documentID, "value": data])
+    }
+
+    // We validate that the SDK orders the vector field the same way as the backend
+    // by comparing the sort order of vector fields from getDocsFromServer and
+    // onSnapshot. onSnapshot will return sort order of the SDK,
+    // and getDocsFromServer will return sort order of the backend.
+
+    let orderedQuery = collection.order(by: "embedding")
+
+    let watchSnapshot = try await Future<QuerySnapshot, Error>() { promise in
+      orderedQuery.addSnapshotListener { snapshot, error in
+        if let error {
+          promise(Result.failure(error))
+        }
+        if let snapshot {
+          promise(Result.success(snapshot))
+        }
+      }
+    }.value
+
+    let getSnapshot = try await orderedQuery.getDocuments(source: .server)
+
+    // Compare the snapshot (including sort order) of a snapshot
+    // from Query.onSnapshot() to an actual snapshot from Query.get()
+    XCTAssertEqual(watchSnapshot.count, getSnapshot.count)
+    for i in 0 ..< min(watchSnapshot.count, getSnapshot.count) {
+      XCTAssertEqual(
+        watchSnapshot.documents[i].documentID,
+        getSnapshot.documents[i].documentID
+      )
+    }
+
+    // Compare the snapshot (including sort order) of a snapshot
+    // from Query.onSnapshot() to the expected sort order from
+    // the backend.
+    XCTAssertEqual(watchSnapshot.count, docs.count)
+    for i in 0 ..< min(watchSnapshot.count, docs.count) {
+      XCTAssertEqual(watchSnapshot.documents[i].documentID, docs[i]["id"] as! String)
+    }
+  }
+
+  func testSdkOrdersVectorFieldSameWayOnlineAndOffline() async throws {
```

Comment:
Can we add one more test to show they filter vector fields the same?

Reply:
This test already covered queries that filter on vector fields. I split this test into two tests to improve the readability.